src/strategies/GifStrategy.d.ts

```diff
@@ -0,0 +1,19 @@
+import {Gif} from "../Gif";
+
+export interface GifStrategy {
```

Comment:
Why not implementing this as a class which can be extended from "strategies" if they don't have a naive implementation? 
Something like:
```ts
export class BaseStrategy<T extends SearchClient & TrendingClient> implements GifStrategy {
    public constructor(
        private readonly name: string;
        private readonly client: T
    ) {}

    public async search(query: string, limit: number): Promise<Array<Gif>> {
        return this.client.search(query, limit);
    }

    public async trending(limit: number): Promise<Array<Gif>> {
        return this.client.trending(limit);
    }
}
```

Reply:
I mentioned it higher up, but since SearchClient and TrendingClient aren't strictly needed, there's no was to provide an abstract/base strategy. The strategy's job it to transform the clients' interface so the clients really can vary a lot.

Reply:
I can see that the SearchClient and TrendingClient interfaces can be confusing. As they're not needed, I'll remove them.